assets/js/blocks/products/all-products/edit.tsx

```diff
@@ -48,51 +47,58 @@ import { getSharedContentControls, getSharedListControls } from '../edit';
 import Block from './block';
 import './editor.scss';
 
-/**
- * Component to handle edit mode of "All Products".
- */
-class Editor extends Component {
-	static propTypes = {
-		/**
-		 * The attributes for this block.
-		 */
-		attributes: PropTypes.object.isRequired,
-		/**
-		 * A callback to update attributes.
-		 */
-		setAttributes: PropTypes.func.isRequired,
-		/**
-		 * From withSpokenMessages.
-		 */
-		debouncedSpeak: PropTypes.func.isRequired,
-	};
+type Attributes = {
+	columns: number;
+	rows: number;
+	alignButtons: boolean;
+	layoutConfig: [ string, object? ][];
+};
 
-	state = {
-		isEditing: false,
-		innerBlocks: [],
-	};
+type Props = {
+	contentVisibility: number;
+	orderby: number;
+	isPreview: number;
+	clientId: string;
+	attributes: Attributes;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
+};
 
-	blockMap = getBlockMap( 'woocommerce/all-products' );
+export default function Edit( props: Props ): JSX.Element {
+	const [ isEditing, setIsEditing ] = useState< boolean >( false );
+	const [ , setInnerBlocks ] = useState< BlockInstance[] | boolean >( false );
+	const blockMap = getBlockMap( 'woocommerce/all-products' );
 
-	componentDidMount = () => {
-		const { block } = this.props;
-		this.setState( { innerBlocks: block.innerBlocks } );
-	};
+	const { clientId, attributes, setAttributes } = props;
+
+	const { columns, rows, alignButtons, layoutConfig } = attributes;
 
-	getTitle = () => {
+	const { innerBlocks } = useSelect( ( select ) => {
+		const { getBlock } = select( 'core/block-editor' );
+		const block = getBlock( clientId );
+		return {
+			innerBlocks: block ? block.innerBlocks : [],
+		};
+	} );
+
+	useEffect( () => {
+		setInnerBlocks( innerBlocks );
+	}, [] );
+
+	const { replaceInnerBlocks } = useDispatch( 'core/block-editor' );
+	const debouncedSpeak = useDebounce( speak );
```

Comment:
Is there a reason we cannot use the `debouncedSpeak` provided through the props as it was before, but rather create a new one?

Reply:
@kmanijak Let me know if I've understood it incorrectly. Previously, `debouncedSpeak` was the props added by the `withSpokenMessages` HOC and I've replaced that with the `useDebounce` hook. Is that not correct?

Reply:
> Previously, debouncedSpeak was the props added by the withSpokenMessages HOC and I've replaced that with the useDebounce hook. Is that not correct?

Oh, I didn't spot that. But I think we could stick to keep using `withSpokenMessages` which is used across multiple blocks. Is there a reason to replace it with a custom implementation?